engine/pkg/meta/client.go

```diff
@@ -14,31 +14,28 @@
 package meta
 
 import (
-	"github.com/pingcap/tiflow/engine/pkg/meta/internal/etcdkv"
-	"github.com/pingcap/tiflow/engine/pkg/meta/internal/etcdkv/namespace"
+	"github.com/pingcap/tiflow/engine/pkg/meta/internal"
 	metaModel "github.com/pingcap/tiflow/engine/pkg/meta/model"
 )
 
-// etcdKVClient is the implement of kv interface based on etcd
-// Support namespace isolation and all kv ability
-// etcdImpl -> kvPrefix+Closer -> etcdKVClient
-type etcdKVClient struct {
-	metaModel.Client
-	metaModel.KV
-	tenantID string
+// NewKVClientWithNamespace return a KVClient with namspace isolation
+func NewKVClientWithNamespace(cc metaModel.ClientConn, projectID metaModel.ProjectID,
+	jobID metaModel.JobID,
+) (metaModel.KVClient, error) {
+	builder, err := internal.GetClientBuilder(cc.ClientType())
+	if err != nil {
+		return nil, err
+	}
+
+	return builder.NewKVClientWithNamespace(cc, projectID, jobID)
 }
 
-// NewPrefixKVClient return a kvclient with namespace
-func NewPrefixKVClient(cli metaModel.KVClientEx, tenantID string) metaModel.KVClient {
-	pfKV := namespace.NewPrefixKV(cli, namespace.MakeNamespacePrefix(tenantID))
-	return &etcdKVClient{
-		Client:   cli,
-		KV:       pfKV,
-		tenantID: tenantID,
+// NewKVClient return a KVClient without namespace isolation
+func NewKVClient(cc metaModel.ClientConn) (metaModel.KVClient, error) {
```

Comment:
This API is used in `engine/pkg/meta/example_test.go` only, can we remove it and use `NewKVClientWithNamespace` in example code.

Reply:
I have replaced the `NewKVClient` with `NewKVClientWithNamespace ` in example code.
But removing the `NewKVClient` may seems strange, a more normal kvclient is created by `NewKVClient`.

Reply:
Yep `NewKVClient` is more general, but what I mean is if we don't use it anywhere, we don't need to keep it.

Reply:
Ok. Done.